### PR TITLE
RWA-113 - Add listener to Camunda to call task configuration service

### PR DESCRIPTION
**Pull request by alexbaeza**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-113

### Change description ###

Added task event listener that calls task-configuration service to retrieve configuration before creating the task in the database.

Added s2s integration to generate service tokens before calling task-configuration.
Added wa-task-configuration integation.
Added FeignRetryPolicy to attempt to get configuration up to 3 times.
Added unit tests and integration test.
Setup integration tests.
Added S2S token generation and secret.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by lewis150193 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/domain/response/ConfigureTaskResponse.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package uk.gov.hmcts.reform.camunda.bpm.domain.response;
+
+import java.util.Map;
+
+public class ConfigureTaskResponse {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/camunda/bpm/domain/response/ConfigureTaskResponse.java:5]: I think Ravi suggested using immutable objects if possible.

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/services/FeignRetryPolicy.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+import feign.FeignException;
+import org.slf4j.Logger;
+
+import java.util.function.Supplier;
+
+import static org.slf4j.LoggerFactory.getLogger;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+public class FeignRetryPolicy<T> {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/camunda/bpm/services/FeignRetryPolicy.java:11]: Looks creative! There is also this spring retry mechanism in place: https://github.com/spring-projects/spring-retry

**Review by iguannago — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/domain/response/ConfigureTaskResponse.java (hunk 1 above)
Comment (alexbaeza) [reply src/main/java/uk/gov/hmcts/reform/camunda/bpm/domain/response/ConfigureTaskResponse.java:5]: on it well spotted

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/config/EventHandlerConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    private static final String TEST_PURPOSES_ASSIGNEE_ID = "demo";
+    private final TaskConfigurationService taskConfigurationService;
+
+    public EventHandlerConfiguration(TaskConfigurationService taskConfigurationService) {
+        this.taskConfigurationService = taskConfigurationService;
+    }
+
+    @EventListener
+    public void onTaskCreatedEvent(DelegateTask delegateTask) {
+        System.out.println("called");
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/camunda/bpm/config/EventHandlerConfiguration.java:25]:
```suggestion

```

**Pull request closed without merging**